easybuild/easyblocks/s/superlu.py

```diff
@@ -0,0 +1,133 @@
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing the SuperLU library, implemented as an easyblock
+
+@author: Xavier Besseron (University of Luxembourg)
+"""
+
+import os
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+from easybuild.tools.modules import get_software_root
+
+
+class EB_SuperLU(CMakeMake):
+    """
+    Support for building the SuperLU library
+    """
+
+    @staticmethod
+    def extra_options():
+        """
+        Define custom easyconfig parameters for SuperLU.
+        """
+        extra_vars = {
+            'build_shared_libs': [False, "Build shared library (instead of static library)", CUSTOM],
+        }
+        return CMakeMake.extra_options(extra_vars)
+
+
+    def configure_step(self):
+        """
+        Set the CMake options for SuperLU
+        """
+        self.cfg['separate_build_dir'] = True
+
+        if self.cfg['build_shared_libs']:
+            self.cfg['configopts'] += "-DBUILD_SHARED_LIBS=ON "
+        else:
+            self.cfg['configopts'] += "-DBUILD_SHARED_LIBS=OFF "
+
+        # Make sure to add -fPIC
+        self.cfg['configopts'] += "-DCMAKE_POSITION_INDEPENDENT_CODE=ON "
+
+        # Make sure not to build the slow BLAS library included in the package
+        self.cfg['configopts'] += "-Denable_blaslib=OFF "
+
+        # Set the BLAS library to use
+        # For this, use the BLA_VENDOR option from the FindBLAS module of CMake
+        # cf https://cmake.org/cmake/help/latest/module/FindBLAS.html
+        if get_software_root('IMKL'):
+            self.cfg['configopts'] += '-DBLA_VENDOR="Intel10_64lp" '
+        elif get_software_root('ACML'):
+            self.cfg['configopts'] += '-DBLA_VENDOR="ACML" '
+        elif get_software_root('ATLAS'):
+            self.cfg['configopts'] += '-DBLA_VENDOR="ATLAS" '
+        elif get_software_root('OpenBLAS'):
+            # Unfortunately, OpenBLAS is not recognized by FindBLAS from CMake,
+            # we have to specify the OpenBLAS library manually
+            self.cfg['configopts'] += '-DBLAS_LIBRARIES="${EBROOTOPENBLAS}/lib/libopenblas.a;-pthread" '
+        else:
+            # Fallback on Generic BLAS otherwise
+            self.cfg['configopts'] += '-DBLA_VENDOR="Generic" '
```

Comment:
it depends on what 'generic' means here...

if this is the reference netlib BLAS, it's a big no-no to just fall back to that and have a 'working' SuperLU as result, since you'll actually be getting a functional yet crippled (performance-wise) install